omniledger/contracts/coins.go

```diff
@@ -58,23 +59,28 @@ func (s safeUint64) sub(a uint64) (safeUint64, error) {
 // You can only delete a contractCoin instance if the account is empty.
 func ContractCoin(cdb omniledger.CollectionView, inst omniledger.Instruction, c []omniledger.Coin) (sc []omniledger.StateChange, cOut []omniledger.Coin, err error) {
 	cOut = c
+
+	var darcID darc.ID
+	_, _, darcID, err = cdb.GetValues(inst.InstanceID.Slice())
+	if err != nil {
+		return
+	}
+
 	switch inst.GetType() {
 	case omniledger.SpawnType:
-		// Spawn creates a new coin account as a separate instance. The subID is
-		// taken from the hash of the instruction.
-		ca := omniledger.InstanceID{
-			DarcID: inst.InstanceID.DarcID,
-			SubID:  omniledger.NewSubID(inst.Hash()),
-		}
-		log.Lvlf3("Spawing coin to %x", ca.Slice())
+		// Spawn creates a new coin account as a separate instance.
+		ca := omniledger.InstanceIDFromSlice(inst.Hash())
+		log.Lvlf3("Spawning coin to %x", ca.Slice())
 		sc = []omniledger.StateChange{
-			omniledger.NewStateChange(omniledger.Create, ca, ContractCoinID, make([]byte, 8)),
+			omniledger.NewStateChange(omniledger.Create, ca,
```

Comment:
Where is the `darcID` stored for this new coin instance?

Reply:
Fixed. That code was a messy leftover that was probably working by accident.